iothub/device/src/Pipeline/AmqpTransportHandler.cs

```diff
@@ -22,8 +22,7 @@ internal class AmqpTransportHandler : TransportHandlerBase
         protected AmqpUnit _amqpUnit;
         private readonly Action<DesiredProperties> _onDesiredStatePatchListener;
         private readonly object _lock = new();
-        private readonly ConcurrentDictionary<string, TaskCompletionSource<AmqpMessage>> _twinResponseCompletions = new();
-        private readonly ConcurrentDictionary<string, DateTimeOffset> _twinResponseTimeouts = new();
+        private readonly ConcurrentDictionary<string, PendingAmqpTwinOperation> _pendingTwinOperations = new();
```

Comment:
Use just one concurrent dictionary for tracking twin operations and the associated time.